camayoc/tests/qcs/api/credentials/test_network_creds.py

```diff
@@ -0,0 +1,192 @@
+# coding=utf-8
+"""Tests for Network source use cases of ``Credential`` API endpoint.
+
+:caseautomation: automated
+:casecomponent: api
+:caseimportance: high
+:caselevel: integration
+:requirement: Sonar
+:testtype: functional
+:upstream: yes
+"""
+from pathlib import Path
+
+import pytest
+
+from camayoc import api
+from camayoc.qcs_models import Credential
+from camayoc.utils import uuid4
+from camayoc.tests.qcs.utils import assert_matches_server
+
+
+def test_update_password_to_sshkeyfile(
+        shared_client, cleanup, isolated_filesystem):
+    """Create a network credential using password and switch it to use sshkey.
+
+    :id: 6e557092-192b-4f75-babc-abc5774fe965
+    :description: Create a network credential with password, then update it
+        to use a sshkey.
+    :steps:
+        1) Create a network credential with a username and password.
+        2) Update the network credential deleting password and adding sshkey.
+        3) Confirm network credential has been updated.
+    :expectedresults: The network credential is updated.
+    """
+    cred = Credential(
+        cred_type='network',
+        client=shared_client,
+        password=uuid4())
+    cred.create()
+    # add the id to the list to destroy after the test is done
+    cleanup.append(cred)
+    assert_matches_server(cred)
+
+    sshkeyfile = Path(uuid4())
+    sshkeyfile.touch()
+
+    cred.ssh_keyfile = str(sshkeyfile.resolve())
+    cred.password = None
+    cred.update()
+    assert_matches_server(cred)
+
+
+def test_update_sshkey_to_password(
+        shared_client, cleanup, isolated_filesystem):
+    """Create a network credential using password and switch it to use sshkey.
+
+    :id: d24a54b5-3d8c-44e4-a0ae-61584a15b127
+    :description: Create a network credential with a sshkey, then update it
+        to use a password.
+    :steps:
+        1) Create a network credential with a username and sshkey.
+        2) Update the network credential deleting sshkey and updating
+           password.
+        3) Confirm network credential has been updated.
+    :expectedresults: The network credential is updated.
+    """
+    ssh_keyfile = Path(uuid4())
+    ssh_keyfile.touch()
+
+    cred = Credential(
+        cred_type='network',
+        client=shared_client,
+        ssh_keyfile=str(ssh_keyfile.resolve()),
+    )
+    cred.create()
+    # add the id to the list to destroy after the test is done
+    cleanup.append(cred)
+    assert_matches_server(cred)
+    cred.client.response_handler = api.echo_handler
```

Comment:
This will change the shared_client to echo_handler since the cred.client is the shared_client. This can cause side effects on the other tests...

Reply:
I will just let this have its own client. Good call!